src/aggregation/bucket/histogram/date_histogram.rs

```diff
@@ -55,42 +66,144 @@ pub struct DateHistogramAggregationReq {
+    /// extend the value range. If the bounds for min or max are not extending the range, the value
+    /// has no effect on the returned buckets.
+    ///
+    /// Cannot be set in conjunction with min_doc_count > 0, since the empty buckets from extended
+    /// bounds would not be returned.
+    pub extended_bounds: Option<HistogramBounds>,
+
     /// Whether to return the buckets as a hash map
     #[serde(default)]
     pub keyed: bool,
 }
 
 impl DateHistogramAggregationReq {
+    pub(crate) fn to_histogram_req(&self) -> crate::Result<HistogramAggregation> {
+        self.validate()?;
+        Ok(HistogramAggregation {
+            field: self.field.to_string(),
+            interval: parse_into_microseconds(&self.fixed_interval)? as f64,
+            offset: self
+                .offset
+                .as_ref()
+                .map(|offset| parse_offset_into_microseconds(offset))
+                .transpose()?
+                .map(|el| el as f64),
+            min_doc_count: self.min_doc_count,
+            hard_bounds: None,
+            extended_bounds: None,
+            keyed: self.keyed,
+        })
+    }
+
     fn validate(&self) -> crate::Result<()> {
+        if self.interval.is_some() {
+            return Err(crate::TantivyError::InvalidArgument(format!(
+                "`interval` parameter {:?} in date histogram is unsupported, only \
+                 `fixed_interval` is supported",
+                self.interval
+            )));
+        }
+        if self.format.is_some() {
+            return Err(crate::TantivyError::InvalidArgument(
+                "format parameter on date_histogram is unsupported".to_string(),
+            ));
+        }
+
+        if self.date_interval.is_some() {
+            return Err(crate::TantivyError::InvalidArgument(
+                "date_interval in date histogram is unsupported, only `fixed_interval` is \
+                 supported"
+                    .to_string(),
+            ));
+        }
+
+        parse_into_microseconds(&self.fixed_interval)?;
+
         Ok(())
     }
 }
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Error)]
 /// Errors when parsing the fixed interval for `DateHistogramAggregationReq`.
 pub enum DateHistogramParseError {
     /// Unit not recognized in passed String
+    #[error("Unit not recognized in passed String {0:?}")]
     UnitNotRecognized(String),
     /// Number not found in passed String
+    #[error("Number not found in passed String {0:?}")]
     NumberMissing(String),
     /// Unit not found in passed String
+    #[error("Unit not found in passed String {0:?}")]
     UnitMissing(String),
+    /// Offset invalid
+    #[error("passed offset is invalid {0:?}")]
+    InvalidOffset(String),
 }
 
-fn parse_into_milliseconds(input: &str) -> Result<u64, DateHistogramParseError> {
+fn parse_offset_into_microseconds(input: &str) -> Result<i64, AggregationError> {
+    let is_sign = |byte| &[byte] == b"-" || &[byte] == b"+";
```

Comment:
Why do we need to parse numbers?

Reply:
ah ok, so you wanted to reuse `parse_into_microseconds`, but offset can be negative or something like that?

Reply:
no really I don't get it... O_o

Reply:
> ah ok, so you wanted to reuse parse_into_microseconds, but offset can be negative or something like that?

Yes, that's it. The sign is optional

e.g. offset: `5d` or `-5d`